napari/layers/base/layer_event_handler.py

```diff
@@ -0,0 +1,26 @@
+from ...utils.event_handler import EventHandler
+from napari.utils.base_interface import BaseInterface
+
+
+class LayerEventHandler(EventHandler):
+    """
+    Event handler for layer specific events. Receives change events made from the data/controls/
+    or visual interface and updates all associated components.
+
+    Ex. ImageLayer components = [qt_image.py, vispy_image.py, image.py]
+    """
+
+    def __init__(self, component: BaseInterface = None):
+        super().__init__(component)
+
+    def on_change(self, event=None):
+        """
+        Process changes made from any interface
+        """
+        name = event.type
+        value = event.value
+        print(f"event: {name}")
+        for component in self.components_to_update:
+            update_method_name = f"_on_{name}_change"
+            update_method = getattr(component, update_method_name)
```

Comment:
Might be good to leave some documentation around this area.  If someone adds a new event, and trips `AttributeError`, they'll come poking around and wonder what to do.  Alternative: catch the error and raise a more helpful message.